task/backend/coordinator/coordinator.go

```diff
@@ -135,7 +135,7 @@ func (c *Coordinator) TaskUpdated(ctx context.Context, from, to *taskmodel.Task)
 		if err := c.sch.Release(sid); err != nil && err != taskmodel.ErrTaskNotClaimed {
 			return err
 		}
-	} else {
+	} else if to.Status == string(taskmodel.TaskActive) {
```

Comment:
What would you think about adding a guard clause like:

```
	if to.Status == from.Status && to.Status == string(taskmodel.TaskInactive) {
		return nil
	}
```

At the top of this method & then simplifying the `if ... else if` logic?

Reply:
Yes, this seems like a nicer approach. I'll update that together with the test case.

Reply:
I have update the PR with your proposed changes. I'm resolving this case - please let me know in case the Reviewer is the one that should resolve the PR's case.